shortlister/model.py

```diff
@@ -161,6 +210,58 @@ def clear_score(applicant: Applicant, criterion: Criterion):
         del applicant.scores[criterion]
 
 
+# text extraction
+
+
+def extract_info_from_text(cleaned_list):
+    """gets the section containing applicant information from extracted text"""
+
+    # fields names to get related applicant information
+    fields = (
+        "First Name",
+        "Last Name",
+        "Email Address",
+        "Preferred Phone Number",
+        "Postcode",
+        "Country & Region",
+    )
+    info = []
+
+    # removes header/footer and other irrelevant info
+    applicant_info = cleaned_list[1:-5]
+    right_to_work = cleaned_list[-5:-1]
+
+    # filter out the field name and retain only the info to applicant
+    for field in fields:
+        for i in applicant_info:
+            if field in i:
+                unclean = i.replace(field, "")
+                clean = re.sub(r"\s{2,}", " ", unclean)
+                info.append(clean.strip())
+
+    # finds where the question is and checks the next index which contains the answer to the question
+    if "Do you have the unrestricted right to work in the UK?" in right_to_work:
+        i = right_to_work.index("Do you have the unrestricted right to work in the UK?")
+        if right_to_work[i + 1] == "No":
+            j = right_to_work.index(
+                "If no, please give details of your VISA requirements"
+            )
+            visa_req_text = right_to_work[j + 1]
+            applicant_right_to_work = False
+
+        elif right_to_work[i + 1] == "Yes":
+            applicant_right_to_work = True
+            visa_req_text = None
+        else:
+            print("Something went wrong")
+        info.append(applicant_right_to_work)
+        info.append(visa_req_text)
+    else:
+        raise ValueError("Right to work is not identified")
```

Comment:
As mentioned above, remove raising of error. It's better to return what we can and leave some fields "unretrievable" if we can't find it in the PDF text.